### ISSUE TITLE
Need Better filters for  Request Sizing Recommendations Section

### ISSUE THREAD

**Issue by wamak9** [kubecost/cost-analyzer-helm-chart#1670]
**What problem are you trying to solve?**
Need filters on the Request Sizing Recommendations under 'Breakdown' when viewing recommendation at cluster level.

**Describe the solution you'd like**
When viewing sizing recommendations based on filters at cluster level. Under breakdown section I see recommendations at container level. so this is very annoying since we have more than 1000 pods and hard to follow the recommendation at container level since most of them are part of deployment.

**Describe alternatives you've considered**
So, for the Breakdown section maybe two options can be introduced.
1) Added a filter to sort by deployment or Service rather than container 
2) When filters with cluster is selected default the Breakdown to Service or Deployment and then proceed to drilldown more based on container if needed.
Extra Feature request:
3) Added Sizing recommendation based on the usage with in the custom data range as well. At the moment if we select 30 days  is it 30 days from when someone is viewing the page ? What if I want recommendation based on the 2 week range on a new cluster ?

**How would users interact with this feature?**
Easy for cluster which has more than 1000 pods.
Drilldown at deployment level when cluster filter is selected gives more visibility into  a service/deployment.
Driller down can be done at Namespace level as well.

**Note:**

Once Cluster filter is added there are options add more filters as well.
But We are talking this from a point of view where  some manager just wants to view the sizing recommendations and at glance so he can see what deployment has efficiency and better recommendation and doesnt have to go through every individual filter to figure and then drill down even more.

If additional information is needed, probably I can setup a session with support team.
cc: keithhand

gz#2510<br><br><i>(related to <a href='https://kubecost.zendesk.com/agent/tickets/2510'>Zendesk ticket #2510</a>)</i>

┆Issue is synchronized with this [Jira Task](https://kubecost.atlassian.net/browse/GIB-168) by [Unito](https://www.unito.io)

**Comment by keithhand** [comment]
Hi @akramwaseem, I think some of this might be achievable today. If you select to use the "Controller" and "Controller Type" options to filter, you should be able to achieve a "deployment filter" so that we can start to view more specific objects.

![image](https://user-images.githubusercontent.com/43491549/187000186-60c2fe58-4550-450c-9a94-c46f5d344dfa.png)

Regarding a "Deployment breakdown," I think the biggest struggle would be that resources are applied/consumed at the pod level. So if we have a deployment containing multiple pods, displaying the usage and recommendation would be challenging. We could combine the usage data; however, I'm not sure if that provides a good "recommendation," as that would generally be set based on the pod. Does that make sense?

I am curious about the third point of us adding a query window. I could see this being useful to validate previous configurations. I'm unsure if a limitation prevents us from doing this now. Tagging @Adam-Stack-PM to see if this is a feature we can look into implementing.

**Comment by Adam-Stack-PM** [comment]
@keithhand Thanks for the tag and @akramwaseem thank you for the idea. Will review for an upcoming release. 

@dwbrown2, Big +1 on for a larger project to bring best in class filtering across the application.

**Comment by Adam-Stack-PM** [comment]
@dwbrown2, Would love your product input on this issue.

**Comment by dwbrown2** [comment]
Thanks guys! Can you share a bit more on why this is annoying? Are these 1000 pods not all part of the same deployment? Are you trying to set different requests across pods within a deployment? How would you be doing that?

You should already be able to filter by cluster, deployment, service and more. Are those not meeting your needs?

+1 on custom date ranges! cc @michaelmdresser as FYI

**Comment by akramwaseem** [comment]
@dwbrown2  -- We have more than 10 teams using the Kubernetes clusters. And more than 100 deployments which are unique on there own. 
If some deployments are new and maybe running for couple of weeks, so to look at the efficiency recommendations, we either have to do 2 days 7 days or 30 days.

How can I look at the whole run period of the deployment (In this case 2 weeks )? This might help get proper recommendations    at deployment level for individual team. Rather than accepting a window which might not make sense and might be off with the recommendations.

**Comment by Adam-Stack-PM** [comment]
@dwbrown2, When you have time wanted to get @akramwaseem response on your radar.

**Comment by dwbrown2** [comment]
> How can I look at the whole run period of the deployment (In this case 2 weeks )?

@akramwaseem here it sounds like interest in custom date ranges, which was just added in new UI as part of 1.99! 

Are there other parts of your request that I'm missing? My apologies if so!

**Comment by dwbrown2** [comment]
Right sizing supports the full range of filters now and will be further improved by autocomplete in the next release. Let us know if there are more feature requests here.
